test/e2e-olm/elasticsearch_test.go

```diff
@@ -256,63 +415,69 @@ func elasticsearchFullClusterTest(t *testing.T, f *framework.Framework, ctx *fra
 		return fmt.Errorf("Timed out waiting for full cluster restart to begin")
 	}*/
 
-	// due to gap mentioned above -- pause here for a few seconds to let the operator do its thing?
-	time.Sleep(10 * time.Second)
-
-	err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, fmt.Sprintf("elasticsearch-cdm-%v-1", dataUUID), 1, retryInterval, timeout)
-	if err != nil {
-		return fmt.Errorf("timed out waiting for Deployment %v: %v", fmt.Sprintf("elasticsearch-cdm-%v-1", dataUUID), err)
-	}
+	t.Log("Waiting for restart after resourse requests/limits update")
 
-	// ensure all prior nodes are ready again
-	err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, fmt.Sprintf("elasticsearch-cdm-%v-1", dataUUID), 1, retryInterval, timeout)
-	if err != nil {
-		return fmt.Errorf("timed out waiting for Deployment %v: %v", fmt.Sprintf("elasticsearch-cdm-%v-1", dataUUID), err)
-	}
+	// Increase restart timeout on full cluster redeploy until min masters available
+	restartTimeout := time.Second * 600
```

Comment:
similar concern here... we could also make sure the pod spec has been updated?

Reply:
As discussed elsewhere, the last acknowledgement was to wait for available pods that include the changed spec. Testing needs to wait for each deployment to get ready vs. available (`status.ReadyReplicas` vs. `AvailableReplicas`). However, the scale down an up of the cluster get's stuck to the point of no-recovery. One can observe both pods (in a two nodes cluster) to fight against each other and eventually crash loop due to missing shards on the security index:

e.g. pod 1:
```
[2020-07-07T10:51:47,832][WARN ][r.suppressed             ] [elasticsearch-ied234ea-cdm-mh917ikj-1] path: /.security/security/roles, params: {index=.security, id=roles, type=security}
org.elasticsearch.action.NoShardAvailableActionException: No shard available for [get [.security][security][roles]: routing [null]]
```

e.g. pod 2:
```
[2020-07-07T10:51:44,862][WARN ][r.suppressed             ] [elasticsearch-ied234ea-cdm-mh917ikj-2] path: /.security/security/roles, params: {index=.security, id=roles, type=security}
org.elasticsearch.action.NoShardAvailableActionException: No shard available for [get [.security][security][roles]: routing [null]] 
```

In summary this needs more time of investigation. Thus I will split the extra spec chenk to a separate PR, because this PR already adds the required reliability to unblock other gated PRs (e.g. #400, #409, #410, #414).